EventEnabledInsurance/readme.md

```diff
@@ -559,3 +559,64 @@ DB_writer bar file: Responsible for Reading messages from the Queue `Quote` and
 
 # Testing the POST calls via APIC
 Instructions to load test the POST call via APIC can be found [here](post-load-test-readme.md).
+
+# Component Downtime Testing
+
+Prereqs:
+1. Configure kafka connectors
+2. Call REST endpoint post & get
+
+## I. Shutting down the db writer integration server
+
+1. Delete the integration server:
+    ```sh
+    oc -n $NAMESPACE get integrationserver ace-db-writer-int-srv-eei -o yaml > ~/dbwriter.yaml
+    oc -n $NAMESPACE delete integrationserver ace-db-writer-int-srv-eei
+    ```
+    The post call will succeed but the message won't be taken off the queue and won't be processed
+2. Recreate integration server:
+    ```sh
+    bash oc apply -f ~/dbwriter.yaml
+    ```
+3. Test post and get (they should work now)
+
+## II. Shutting down the queue manager
+
+1. Delete the queue manager instance:
+    ```sh
+    oc -n $NAMESPACE get queuemanager mq-eei -o yaml > ~/eei-queuemanager.yaml
+    oc -n $NAMESPACE delete queuemanager mq-eei
+    ```
+2. Test post call and you should receive an error that contains: `Failed to make a client connection to queue manager`. The get call will still return existing data if the projection claims db has already been populated.
+3. Recreate queue manager and wait for phase to be running:
+    ```sh
+    oc apply -n $NAMESPACE -f ~/eei-queuemanager.yaml
+    oc get queuemanager -n $NAMESPACE mq-eei
+    ```
+4. Test post and get (they should work now)
+
+## III. Shutting down access to postgresql db
+
+1. Setup some env vars:
+    ```sh
+    POSTGRES_NAMESPACE=postgres
+    DB_POD=$(oc get pod -n ${POSTGRES_NAMESPACE} -l name=postgresql -o jsonpath='{.items[].metadata.name}')
+    DB_NAME=$(oc get secret eei-postgres-replication-credential -o json | \
+    jq -r '.data["connector.properties"]' | base64 --decode | grep dbName | awk '{print $2}')
+    ```
+2. Get a psql prompt for the database:
+    ```sh
+    oc exec -n ${POSTGRES_NAMESPACE} -it $DB_POD -- psql -d ${DB_NAME}
+    ```
+3. To simulate the shut down:
+    ```sql
+    REVOKE ALL PRIVILEGES ON QUOTES FROM cp4i_sor_eei;
+    REVOKE ALL PRIVILEGES ON QUOTES FROM cp4i_sor_replication_eei;
+    ```
+4. Post requests will succeed, however the new claim will not show up in the sor db or the projection claims db. Existing claims should still accessible with the get request but everything else will result in a 404.
+5. To restart postgres, make sure you're exec'd into the database and run the following commands:
+    ```sql
+    GRANT ALL PRIVILEGES ON TABLE quotes TO cp4i_sor_eei;
+    GRANT ALL PRIVILEGES ON TABLE quotes TO cp4i_sor_replication_eei;
+    ```
+6. Once the permissions have been restored, the new claim should show up in the psql db as well as the projection claims app.
```

Comment:
Does the backout queue need to be doc'd here somehow? What happens if the time between 3 and 5 above is long enough for messages to be put in the backout queue? And how long is that? Since DB Writer pulls a message every 1 second I guess it'll take around 99 seconds until message start appearing in the backout queue?

Reply:
I've doc'd the backout queue behaviour :)